backend/src/question/question.service.ts

```diff
@@ -123,23 +246,29 @@ export class QuestionService {
   async getSummariesFromTopicTags(topicTags: string[], matchType: string) {
     const allTopicTags = await this.getAllTopics();
     const validTopicTagArray = _.intersection(allTopicTags, topicTags);
+    const cachedSummariesFromTopicTag: QuestionSummaryFromDb[][] = [];
 
-    const validSummaries = await this.prisma.topicTag.findMany({
-      where: { topicSlug: { in: validTopicTagArray } },
-      select: {
-        questionSummaries: {
-          select: QUESTION_SUMMARY_SELECT,
-        },
-      },
-    });
+    let cachedSummaries = await this.cache.getKeyInNamespace<
+      QuestionSummaryFromDb[]
+    >([NAMESPACES.QUESTIONS], QUESTION_SUMMARIES);
 
-    // Array questions grouped by matched valid topic tags
-    const flatValidSummaries: QuestionSummaryFromDb[][] = validSummaries.map(
-      (v) => v.questionSummaries
-    );
+    if (!cachedSummaries) {
+      this.getAllSummaries();
+      cachedSummaries = await this.cache.getKeyInNamespace<
+        QuestionSummaryFromDb[]
+      >([NAMESPACES.QUESTIONS], QUESTION_SUMMARIES);
+    }
+
+    for (const tag in validTopicTagArray) {
+      cachedSummariesFromTopicTag.push(
+        cachedSummaries.filter((summary) =>
+          summary.topicTags.map((topic) => topic.topicSlug).includes(tag)
+        )
+      );
+    }
```

Comment:
```suggestion
    for (const tag in validTopicTagArray) {
      const currentMatchedQuestions = cachedSummaries.filter((summary) => {
        const currentQuestionTags = summary.topicTags.map((topic) => topic.topicSlug)
        return currentQuestionTags.includes(tag);
      })
      cachedSummariesFromTopicTag.push(currentMatchedQuestions);
    }
```

Improve readability, not sure if this snippet works since